src/Routes/RosPage/RosPage.js

```diff
@@ -79,7 +81,11 @@ class RosPage extends React.Component {
         };
 
         let url = new URL(ROS_API_ROOT + SYSTEMS_API_ROOT,  window.location.origin);
-        url.search = new URLSearchParams(params).toString();
+        let query = new URLSearchParams(params);
+        fetchParams?.stateFilter?.length > 0 && fetchParams.stateFilter.map((stateFilterValue) => {
+            query.append('state', stateFilterValue);
+        });
```

Comment:
No need for extra check, also would be better to use `forEach` if we are not consuming the value from `map`
```suggestion
        fetchParams?.stateFilter?.forEach((stateFilterValue) => {
            query.append('state', stateFilterValue);
        });
```

Reply:
Thank you for the suggestion @karelhala, updated the code with same 👍🏻